sw/device/lib/runtime/epmp.h

```diff
@@ -0,0 +1,458 @@
+ * must be a power of two greater than four. The `region` (both start and end
+ * addresses) must also be aligned to the same power of two.
+ *
+ * If `kEpmpGranularity` is greater than zero then the entire `region` must
+ * also be aligned to `2 ** (2 + kEpmpGranularity)`.
+ *
+ * Example:
+ *
+ *   ...
+ *   epmp_transaction_t t;
+ *   res0 = epmp_transaction_init(&t, kEpmpEntryCount2);
+ *   res1 = epmp_entry_configure_napot(&t, kEpmpEntry0,
+ *              (epmp_region_t){ .start = 0x10, .end = 0x20 },
+ *              kEpmpPermissionsUnlockedMachineAllUserAll);
+ *   res2 = epmp_entry_configure_napot(&t, kEpmpEntry1,
+ *              (epmp_region_t){ .start = 0x50, .end = 0x58 },
+ *              kEpmpPermissionsUnlockedMachineAllUserNone);
+ *   res3 = epmp_transaction_push(&t, kEpmpEntryCount2);
+ *   ...
+ *
+ * Result:
+ *
+ *   Entry | Value of `pmpaddr`        | Value of `pmpcfg` |
+ *   ======+===========================+===================+
+ *       0 | 0x41 ((0x10 >> 2) | 0b01) |         0b0011111 |
+ *       1 | 0x14 ((0x50 >> 2) | 0b00) |         0b0011000 |
+ *
+ * @param state State to update (unchanged on error).
+ * @param entry Entry index to update (0 <= `entry` < `kEpmpNumRegions`)
+ * @param region Region start and end addresses.
+ * @param permissions Updated permissions to write to pmpcfg for `entry`.
+ * @return `epmp_entry_configure_result_t`.
+ */
+epmp_entry_configure_result_t epmp_entry_configure_napot(
+    epmp_transaction_t *transaction, epmp_entry_t entry, epmp_region_t region,
+    epmp_permissions_t permissions);
+
+/**
+ * Enable or disable the Rule Locking Bypass (RLB) feature.
+ *
+ * When enabled (mseccfg.RLB=1) the Rule Locking Bypass features allows
+ * locked entries to be modified. If any PMP entries are locked and RLB
+ * is disabled (mseccfg.RLB=0) then it is no longer possible to enable
+ * RLB. RLB will be set as requested or an error will be returned.
+ *
+ * @param toggle Whether RLB should be enabled or disabled.
+ * @return `epmp_result_t`
+ */
+epmp_result_t epmp_set_rule_locking_bypass(epmp_rlb_toggle_t toggle);
+
+/**
+ * Get the status of the Rule Locking Bypass (RLB) feature.
+ *
+ * When enabled (mseccfg.RLB=1) the Rule Locking Bypass features allows
+ * locked entries to be modified. If any PMP entries are locked and RLB
+ * is disabled (mseccfg.RLB=0) then it is no longer possible to enable
+ * RLB. RLB will be set as requested or an error will be returned.
+ *
+ * TODO: only for debugging?
+ *
+ * @param[out] toggle Whether RLB is enabled or disabled.
+ * @return `epmp_result_t`
+ */
+epmp_result_t epmp_get_rule_locking_bypass(epmp_rlb_toggle_t *toggle);
+
+/**
+ * Read the current EPMP configuration and check it against the shadow
+ * copy stored in RAM.
+ *
+ * @param[out] state Destination to write register values to (optional).
+ * @return `epmp_result_t`
+ */
+epmp_result_t epmp_get(epmp_state_t *state);
```

Comment:
"Get" what? Maybe be clearer on what is being gotten, e.g. `get_state`.

Reply:
Or even `epmp_validate_state`, `epmp_state_is_valid` or something similar?